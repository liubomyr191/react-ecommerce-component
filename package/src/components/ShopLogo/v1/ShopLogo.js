import React, { Component } from "react";
import PropTypes from "prop-types";

export default class ShopLogo extends Component {
  static propTypes = {
    /**
     * The primary shop's logo url
     */
    shopLogoUrl: PropTypes.string,
    /**
     * The primary shop's name
     */
    shopName: PropTypes.string.isRequired
  }

  render() {
    const { shopLogoUrl, shopName } = this.props;

    return (
      <div>
        {
          shopLogoUrl ? (
            <img src={shopLogoUrl} alt={shopName} />
          ) : (
            shopName
          )
        }
      </div>
    );
  }
}

