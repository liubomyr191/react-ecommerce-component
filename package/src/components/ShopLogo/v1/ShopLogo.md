### Overview
Renders a shop's logo if a logo URL is provided, otherwise, it will render the shop's name.

#### Usage

##### Default

```jsx
<ShopLogo shopLogoUrl="//placehold.it/60" shopName="Reaction" />
```

##### Without a logo

```jsx
<ShopLogo shopName="Reaction" />
```
