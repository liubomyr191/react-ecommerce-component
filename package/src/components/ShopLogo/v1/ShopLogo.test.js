import React from "react";
import renderer from "react-test-renderer";
import ShopLogo from "./ShopLogo";

test("basic snapshot", () => {
  const component = renderer.create((
    <ShopLogo shopLogoUrl="http://placehold.it/60" shopName="Reaction" />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
