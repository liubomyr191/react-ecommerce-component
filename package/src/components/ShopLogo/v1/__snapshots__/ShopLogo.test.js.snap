// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  color: #505558;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 24px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.25;
}

<div
  className="c0"
>
  <img
    alt="Reaction"
    src="//placehold.it/60"
  />
</div>
`;
